File: app/src/main/java/com/paycraft/ticketbooker/services/APIService.java
package com.paycraft.ticketbooker.services;

import com.paycraft.ticketbooker.models.ConfirmBooking;
import com.paycraft.ticketbooker.models.OTP;
import com.paycraft.ticketbooker.models.Stations;
import com.paycraft.ticketbooker.models.VerifyOTP;

import java.util.ArrayList;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;


public interface APIService {

    String API_VERSION = "api/v1";

    @GET(API_VERSION + "/stations")
    Observable<ArrayList<Stations>> getStations();

    @POST(API_VERSION + "/otp")
    Observable<OTP> getOTP(@Query("docId") String id);

    @POST(API_VERSION + "/verify_otp/{docId}/{otp}")
    Observable<VerifyOTP> verifyOTP(@Path("docId") String docId, @Path("otp") int otp);

    @POST(API_VERSION+ "/confirmBooking")
    Observable<ConfirmBooking> confirmBooking(@Query("docId") String id,@Query("destId") int dest_id);

}

File: app/src/main/java/com/paycraft/ticketbooker/services/APIService.kt
package com.paycraft.ticketbooker.services

import com.paycraft.ticketbooker.models.ConfirmBooking
import com.paycraft.ticketbooker.models.OTP
import com.paycraft.ticketbooker.models.Stations
import com.paycraft.ticketbooker.models.VerifyOTP
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query
import java.util.*


interface APIService {

    companion object {
        const val API_VERSION = "api/v1";

    }
    @GET("$API_VERSION/stations")
    fun stations(): Observable<ArrayList<Stations>>

    @POST("$API_VERSION/otp")
    fun getOTP(@Query("docId") id: String): Observable<OTP>

    @POST("$API_VERSION/verify_otp/{docId}/{otp}")
    fun verifyOTP(@Path("docId") docId: String, @Path("otp") otp: Int): Observable<VerifyOTP>

    @POST("$API_VERSION/confirmBooking")
    fun confirmBooking(@Query("docId") id: String, @Query("destId") dest_id: Int): Observable<ConfirmBooking>



}